src/components/App.spec.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import { render, fireEvent, wait } from '@testing-library/react'
+
+import App from 'components/App'
+import AppLike from 'test/AppLike'
+import Links from 'ducks/client/links'
+
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+import CozyClient from 'cozy-client'
+import CozyLink from 'cozy-client/dist/CozyLink'
+
+// Mock otherwise we have a fetch is not defined error
+// due to pouchdb-browser. Here we are not concerned with this
+// component
+jest.mock('cozy-pouch-link', () => () => null)
+jest.mock('components/AppSearchBar', () => () => null)
+jest.mock('ducks/commons/Nav', () => () => null)
+jest.mock('lodash/throttle', () => ({
+  default: jest.fn(fn => fn),
+  __esModule: true
+}))
+jest.mock('lodash/throttle')
+
+jest.mock('ducks/client/links', () => ({
+  getActivatePouch: jest.fn()
+}))
+
+const DumbComponent = ({ client }) => {
+  const fetch = async () => await client.query({ doctype: 'io.cozy.fake' })
+  return <button onClick={fetch}>Fetch</button>
+}
+
+describe('App', () => {
+  const requestHandler = jest.fn()
+  const link = new CozyLink(requestHandler)
+  beforeEach(() => {
+    jest.spyOn(Alerter, 'error')
+    jest.spyOn(window.location, 'reload')
+  })
+
+  afterEach(() => {
+    window.location.reload.mockReset()
+  })
+
+  const setup = () => {
+    const client = new CozyClient({ links: [link] })
+    const root = render(
+      <AppLike client={client}>
+        <App>
+          <DumbComponent client={client} />
+        </App>
+      </AppLike>
+    )
+
+    return root
+  }
+
+  describe('App', () => {
+    it('should no show error if pouch is activated', async () => {
+      const error = new TypeError('Failed to fetch')
+      requestHandler.mockReturnValue(Promise.reject(error))
+
+      Links.getActivatePouch = jest.fn().mockImplementation(() => true)
+
+      const root = setup()
+
+      fireEvent.click(root.getByText('Fetch'))
+
+      await wait(() => {
+        expect(Alerter.error).not.toHaveBeenCalled()
+      })
+    })
+
+    it('should show an error and reload page when a request failed', async () => {
+      const error = new TypeError('Failed to fetch')
+      requestHandler.mockReturnValue(Promise.reject(error))
+
+      Links.getActivatePouch = jest.fn().mockImplementation(() => false)
+      window.location.reload = jest.fn()
+
+      const root = setup()
+
+      fireEvent.click(root.getByText('Fetch'))
+
+      await wait(() => {
+        expect(Alerter.error).toHaveBeenCalled()
+
+        const errorMsg = Alerter.error.mock.calls[0][0]
+        expect(errorMsg).toBe(
+          'Connection lost. Try again later or verify your connection and reload the page.'
+        )
```

Comment:
nit: you should avoid testing the translated string. You should only rely on the key.